### PR TITLE
add slug prefix and absolute routes

### LINKED ISSUE #955: Changing shop name breaks urls
For bugs, please fill out the three following sections completely

### Expected behavior
Changing shop name in admin section should change urls in a manner that does not break urls.

### Actual Behavior
When changing the site name in the admin panel to a name containing spaces, urls do not work.

The workaround I've been using is to change the name of the shop in the admin UI to a url friendly one("my_shop_name") and then manually enter the correct display name ("My Shop Name") in the brand.html file.

### PR DESCRIPTION
**Pull request by aaronjudd**
- issue #1022
-allow “absolute” url definition, registry routes without “/“ will be
treated as absolute
- resolves #955 with added getSlug
- related to #1006

### CODE REVIEW

File: packages/reaction-router/common/init.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ const getRegistryRouteName = (packageName, registryItem) => {
  */
 ReactionRouter.initPackageRoutes = () => {
   const pkgs = ReactionCore.Collections.Packages.find().fetch();
-  const prefix = ReactionCore.getShopName().toLowerCase(); // todo add shopId
+  const prefix = getSlug(ReactionCore.getShopName().toLowerCase()); // todo add shopId
```
Comment (zenweasel) [inline packages/reaction-router/common/init.js]: Should we create a global `getShopSlug`? Asking for a friend.

Comment (aaronjudd) [inline packages/reaction-router/common/init.js]: well at this point, getSlug is a global function, and currently, this is the only place that the shop name would be useful slugified - I think.

Comment (zenweasel) [inline packages/reaction-router/common/init.js]: Except you know, if somebody had to create a `return_url` for something. Just speculating.

**Comment by zenweasel** [comment]
Approved.

File: packages/reaction-router/common/init.js (hunk 1 above)
Comment (aaronjudd) [inline packages/reaction-router/common/init.js]: maybe instead we need our own absoluteUrl method..

**Pull request merged**